src/bounds.rs

```diff
@@ -35,6 +35,19 @@ impl Bounds {
         }
     }
 
+    /// Dual method to `union`, which keeps the most restrictive upper-bound of the
+    /// two Bounds values rather than the least.
+    pub fn intersection(lhs: Bounds, rhs: Bounds) -> Bounds {
+        Bounds {
+            min: usize::min(lhs.min, rhs.min),
```

Comment:
This seems a little misleading, wouldn't a true intersection take the max of the min values, as well as the min of the max values? Perhaps this needs a more specific name (intersect_upper or intersection_max or something). Where specifically is it used?

Reply:
This is a bit of an artifact, from `LengthLimited` which ended up getting axed early. The intention was to say:

$$\mathrm{MatchBounds}(\verb!LengthLimited!(\mathit{X}, \mathit{F}) \coloneqq \verb!Bounds::intersection!(\verb!Expr::bounds!(\mathit{X}), \verb!Format::match_bounds!(\mathit{F}))$$

But that never got surfaced as LengthLImited was a bit complicated to implement properly and wasn't ultimately necessary for the PNG case we were working on because RepeatBetween was enough,

Reply:
Right, can always rename it or comment it out as appropriate.